modules/packages/UnrolledLinkedList.chpl

```diff
@@ -0,0 +1,1249 @@
+
+    /*
+      Clear the contents of this unrolledLinkedList.
+
+      .. warning::
+
+        Clearing the contents of this unrolledLinkedList will invalidate all existing
+        references to the elements contained in this unrolledLinkedList.
+    */
+    proc ref clear() {
+      on this {
+        _enter();
+
+        _destroy();
+        _head = nil;
+        _tail = nil;
+        _size = 0;
+
+        _sanity(_size == 0);
+        _sanity(_head == nil);
+
+        _leave();
+      }
+    }
+
+    /*
+      Return a zero-based index into this unrolledLinkedList of the first item whose value
+      is equal to `x`. If no such element can be found this method returns
+      the value `-1`.
+
+      .. warning::
+
+        Calling this method on an empty unrolledLinkedList or with values of `start` or 
+        `end` that are out of bounds will cause the currently running program
+        to halt. If the `--fast` flag is used, no safety checks will be
+        performed.
+
+      :arg x: An element to search for.
+      :type x: `eltType`
+
+      :arg start: The start index to start searching from.
+      :type start: `int`
+
+      :arg end: The end index to stop searching at. A value less than
+                `0` will search the entire unrolledLinkedList.
+      :type end: `int`
+
+      :return: The index of the element to search for, or `-1` on error.
+      :rtype: `int`
+    */
+    proc const indexOf(x: eltType, start: int=0, end: int=-1): int {
+      if boundsChecking {
+        const msg = " index for \"unrolledLinkedList.indexOf\" out of bounds: ";
+
+        if end >= 0 && !_withinBounds(end) then
+          boundsCheckHalt("End" + msg + end:string);
+
+        if !_withinBounds(start) then
+          boundsCheckHalt("Start" + msg + start:string);
+      }
+
+      param error = -1;
+
+      if end >= 0 && end < start then
+        return error;
+
+      var result = error;
+
+      on this {
+        _enter();
+
+        const stop = if end < 0 then _size-1 else end;
+
+        var i = 0;
+        for item in this {
+          if i > stop then break;
+          if i >= start {
+            if x == item {
+              result = i;
+              break;
+            }
+          }
+          i += 1;
+        }
+
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Return the number of times a given element is found in this unrolledLinkedList.
+
+      :arg x: An element to count.
+      :type x: `eltType`
+
+      :return: The number of times a given element is found in this unrolledLinkedList.
+      :rtype: `int`
+    */
+    proc const count(x: eltType): int {
+      var result = 0;
+
+      on this {
+        _enter();
+
+        var count = 0;
+
+        for item in this do
+          if x == item then
+            count += 1;
+
+        result = count;
+
+        _leave();
+      }
+
+      return result;
+    }
+
+    pragma "no doc"
+    proc ref sort(comparator: ?rec=Sort.defaultComparator) {
```

Comment:
You can even make the argument fully generic, if you want to.

Reply:
It's a time ago so I don't remember what exactly drives me to write this line. But I suppose it could be the idea to unify the unrolled linked list and list. And yes, I think I should remove this function and let the compiler do its job.